### DISCUSSION TITLE
Twin 2.0 changes discussion

### DISCUSSION THREAD

**Discussion by ben-rogerson** [ben-rogerson/twin.macro#170] [General] [answered]
**v2 is now out, let's keep this discussion going if you've got any questions about the changes made**

This is the place to discuss plans for Twin v2. 
As it's a major version, ideas that involve breaking changes are definitely up for discussion.

Changes:

- [x] (For 'styled-components') Change the default `styled` import in the styled-components preset to allow configuration
#207

- [x] (For 'styled-components') Set `autoCssProp: true` by default. Now that this feature has been tested, it's time to activate it by default.
#209

- [x] Maintain the ordering for tw and css added on jsx elements. Currently, the tw prop trumps the css prop.
#208 

- [x] Update the default emotion preset to use the new emotion@11 import ~~+ add a new preset for pre-v11 imports~~.
[dd4ce55](https://github.com/ben-rogerson/twin.macro/commit/dd4ce55bf76a291a7d34d71eeb15d5e93654b416#diff-8f688d773f26458c4fe4270eabc365e4fd6833a86ba0a31283215a1ad87e913d)

- [x] Rename the config item 'debugProp' to `dataTwProp` and turn it on by default. The renaming helps distance the option from the other `debug` prefixed options that write to your terminal.
#210

- [x] Make your specified imports for `styled`, `css` and `global` override the imports defined in the `preset`.
#211

**Comment by fvanwijk** [comment]
LGTM!

**Comment by cheapsteak** [comment]
> Maintain the ordering for tw and css added on jsx elements

to clarify, does this mean that `<div tw="bg-blue-100" css={tw`bg-red-100`}/>` is currently resolving to `bg-red-100`, and will in 2.0 remain that way, and `<div css={tw`bg-red-100`} tw="bg-blue-100"/>` is currently resolving to `bg-red-100` and will in 2.0 resolve to `bg-blue-100`?

**Reply by ben-rogerson** [reply]
I had it the other way around, the tw prop trumps the css prop. 

```js
// Current behaviour
<div tw="bg-blue-100" css={[tw`bg-red-100`]}/> // bg will be blue here (v2 will be red)
```

Check out the [demo](https://codesandbox.io/s/react-styled-components-tailwind-twin-starter-forked-7fvo7?autoresize=1&fontsize=14&hidenavigation=1&module=%2Fsrc%2FApp.js&theme=dark) which shows the behaviour.

**Reply by cheapsteak** [reply]
oh yup, this was surprising
would be a fantastic update!

**Comment by ben-rogerson** [comment, marked as answer]
All done here, enjoy v2 everyone!
